tests/alternative/Network.lua

```diff
@@ -1,155 +1,352 @@
+				}
+			end
+
+			unitTest:assertError(error_func, incompatibleTypeMsg("target", "CellularSpace", 2))
+
+			error_func = function()
+				Network{
+					lines = roads,
+					target = communities,
+					inside = 2,
+					outside = function(distance) return distance * 2 end
+				}
+			end
+
+			unitTest:assertError(error_func, incompatibleTypeMsg("inside", "function", 2))
+
+			error_func = function()
+				Network{
+					lines = roads,
+					target = communities,
+					inside = function(distance) return distance end,
+					outside = 2
+				}
+			end
+
+			unitTest:assertError(error_func, incompatibleTypeMsg("outside", "function", 2))
+
+			error_func = function()
+				Network{
+					lines = roads,
+					target = communities,
+					inside = function(distance) return distance end,
+					outside = function(distance) return distance * 2 end,
+					error = "error"
+				}
+			end
+
+			unitTest:assertError(error_func, incompatibleTypeMsg("error", "number", "error"))
+		end
+
+		argumentErrorTests()
 
 		local error_func = function()
-			Network{
-				lines = 2,
-				target = communities,
-				inside = function(distance) return distance end,
-				outside = function(distance) return distance * 2 end
+			local roads = CellularSpace{
+				file = filePath("error/roads-invalid.shp", "gpm"),
+				missing = 0
 			}
-		end
 
-		unitTest:assertError(error_func, incompatibleTypeMsg("lines", "CellularSpace", 2))
+			local communities = CellularSpace{
+				file = filePath("communities.shp", "gpm")
+			}
 
-		error_func = function()
 			Network{
-				lines = communities,
+				lines = roads,
 				target = communities,
+				progress = false,
 				inside = function(distance) return distance end,
 				outside = function(distance) return distance * 2 end
 			}
 		end
 
-		unitTest:assertError(error_func, "Argument 'lines' should be composed by lines, got 'MultiPoint'.")
+		unitTest:assertError(error_func, "Line '7' does not touch any other line. The minimum distance found was: 843.46359196883. If the distance is small, set the error argument, otherwise, correct the line.")
 
 		error_func = function()
-			Network{
-				lines = roads,
-				target = 2,
-				inside = function(distance) return distance end,
-				outside = function(distance) return distance * 2 end
+			local communities = CellularSpace{
+				file = filePath("communities.shp", "gpm")
 			}
-		end
 
-		unitTest:assertError(error_func, incompatibleTypeMsg("target", "CellularSpace", 2))
+			local roads = CellularSpace{
+				file = filePath("error/".."roads_overlay_points.shp", "gpm"),
+				missing = 0
+			}
 
-		error_func = function()
 			Network{
 				lines = roads,
 				target = communities,
-				inside = 2,
+				progress = false,
+				inside = function(distance) return distance end,
 				outside = function(distance) return distance * 2 end
 			}
 		end
 
-		unitTest:assertError(error_func, incompatibleTypeMsg("inside", "function", 2))
+		unitTest:assertError(error_func, "Lines '6' and '14' cross each other.")
```

Comment:
Why it is not possible to use lines that cross each other?

Reply:
The lines are connected by the adjacency relation which they are made by the endpoints distance. Then if the lines cross each other, it is not possible to create that relation.
I think that the really problem happens when the lines share a point where they cross, but if the lines only cross without share a point, it shouldn't have any problem.

Anyway, who verify if line crosses is TerraLib. We would have to refactoring the validation function to work for us.

Note: I don't work in this feature, it already existed before my refactoring.

Reply:
OK